diff --git a/makefile b/makefile
index 5a3aa2f..b6862d1 100644
--- a/makefile
+++ b/makefile
@@ -60,7 +60,7 @@ GUI_SRC = ./src/BAngrGUI.cpp
 OBJ_EXT = .so
 DSP_OBJ = $(DSP)$(OBJ_EXT)
 GUI_OBJ = $(GUI)$(OBJ_EXT)
-B_OBJECTS = $(addprefix $(BUNDLE)/, $(DSP_OBJ) $(GUI_OBJ))
+B_OBJECTS = $(addprefix $(BUNDLE)/, $(DSP_OBJ))
 ROOTFILES = \
 	*.ttl \
 	LICENSE
@@ -116,7 +116,7 @@ ifeq ($(shell $(PKG_CONFIG) --exists 'cairo >= 1.12.0' || echo no), no)
   $(error cairo >= 1.12.0 not found. Please install cairo >= 1.12.0 first.)
 endif
 
-$(BUNDLE): clean $(DSP_OBJ) $(GUI_OBJ)
+$(BUNDLE): clean $(DSP_OBJ)
 	@cp $(ROOTFILES) $(BUNDLE)
 	@mkdir -p $(BUNDLE)/inc
 	@cp $(INCFILES) $(BUNDLE)/inc
